api/v1alpha4/azurecluster_default_test.go

```diff
@@ -803,6 +803,253 @@ func TestNodeOutboundLBDefaults(t *testing.T) {
 				Spec: AzureClusterSpec{
 					NetworkSpec: NetworkSpec{
 						APIServerLB: LoadBalancerSpec{Type: Public},
+						Subnets: Subnets{
+							{
+								Role:          SubnetControlPlane,
+								Name:          "control-plane-subnet",
+								SecurityGroup: SecurityGroup{},
+								RouteTable:    RouteTable{},
+							},
+							{
+								Role:          SubnetNode,
+								Name:          "node-subnet",
+								SecurityGroup: SecurityGroup{},
+								RouteTable:    RouteTable{},
+							},
+						},
+					},
+				},
+			},
+			output: &AzureCluster{
+				ObjectMeta: v1.ObjectMeta{
+					Name: "cluster-test",
+				},
+				Spec: AzureClusterSpec{
+					NetworkSpec: NetworkSpec{
+						Subnets: Subnets{
+							{
+								Role:          SubnetControlPlane,
+								Name:          "control-plane-subnet",
+								SecurityGroup: SecurityGroup{},
+								RouteTable:    RouteTable{},
+							},
+							{
+								Role:          SubnetNode,
+								Name:          "node-subnet",
+								SecurityGroup: SecurityGroup{},
+								RouteTable:    RouteTable{},
+							},
+						},
+						APIServerLB: LoadBalancerSpec{
+							Type: Public,
+						},
+						NodeOutboundLB: &LoadBalancerSpec{
+							Name: "cluster-test",
+							SKU:  SKUStandard,
+							FrontendIPs: []FrontendIP{{
+								Name: "cluster-test-frontEnd",
+								PublicIP: &PublicIPSpec{
+									Name: "pip-cluster-test-node-outbound",
+								},
+							}},
+							Type:                 Public,
+							FrontendIPsCount:     to.Int32Ptr(1),
+							IdleTimeoutInMinutes: to.Int32Ptr(DefaultOutboundRuleIdleTimeoutInMinutes),
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "NAT Gateway enabled - no LB",
```

Comment:
This UT case includes a vanilla configuration (one control plane subnet and one node pool subnet) with NAT Gateway enabled on the node pool. We should expect to *not* get an outbound LB here.